RiscVPkg/Include/Library/OpenSbiPeim.h

```diff
@@ -0,0 +1,340 @@
+/** @file
+  Defines the PEI_RISCV_OPENSBI_LEGACY_PPI to let PEIMs call OpenSBI
+
+Copyright (c) 2020, Hewlett Packard Development LP. All rights reserved.<BR>
+
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef _PEI_RISCV_OPENSBI_PPI_H_
+#define _PEI_RISCV_OPENSBI_PPI_H_
+
+#include <sbi/sbi_scratch.h>
+#include <sbi/sbi_platform.h>
+
+///
+/// Global ID for the PEI_RISCV_OPENSBI_BASE_PPI_GUID.
+///
+#define PEI_RISCV_OPENSBI_BASE_PPI_GUID  \
+  { \
+    0x9af5b3c3, 0x1e8c, 0x4219, { 0xa4, 0xfb, 0x3f, 0xb8, 0xb2, 0xb8, 0xfe, 0xa1 } \
+  }
+
+///
+/// Forward declaration for PEI_RISCV_OPENSBI_BASE_PPI.
+///
+typedef struct _PEI_RISCV_OPENSBI_BASE_PPI PEI_RISCV_OPENSBI_BASE_PPI;
+
+///
+/// SpecVersion:
+/// The minor number of the SBI specification is encoded in the low 24 bits,
+/// with the major number encoded in the next 7 bits.  Bit 32 must be 0 and is
+/// reserved for future expansion.
+///
+typedef
```

Comment:
Earlier comment by @changab 
> We need the complete function header for each prototype. This will be the same header you have for functions in the source file.

I saw the typedefs in other PPIs. Should I remove these here and put a normal header instead?

Reply:
Put header like below for the function prototype, use the same header for the function in C file.
And we don't use typedef for the function prototype.

/**	Brief	description	of	this	function's	purpose.
		Follow	it	immediately	with	the	detailed	description.
		@param[in]							Arg1		Description	of	Arg1.
		@param[in]							Arg2		Description	of	Arg2	This	is	complicated	and	requires
																									multiple	lines	to	describe.
		@param[out]						Arg3		Description	of	Arg3.
		@param[in,	out]		Arg4		Description	of	Arg4.
		@retval		VAL_ONE		Description	of	what	VAL_ONE	signifies.
		@retval		OTHER				This	is	the	only	other	return	value.	If	there	were	other
																				return	values,	they	would	be	listed.
**/

Reply:
Ah you mean only the comments in front of the typedef is missing?

Reply:
Yes. Only the comments in front of typdef is missed.